src/Utils/request/useTanStackQueryInstead.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery } from "@tanstack/react-query";
+
+import request from "@/Utils/request/request";
+import {
+  QueryRoute,
+  RequestOptions,
+  RequestResult,
+} from "@/Utils/request/types";
+
+import { mergeRequestOptions } from "./utils";
+
+export interface QueryOptions<TData> extends RequestOptions<TData> {
+  prefetch?: boolean;
+  refetchOnWindowFocus?: boolean;
+  key?: string;
+}
+
+export default function useTanStackQueryInstead<TData>(
+  route: QueryRoute<TData>,
+  options?: QueryOptions<TData>,
+) {
+  const { data: response, isLoading } = useQuery({
+    queryKey: [route.path, options?.pathParams, options?.query],
+    queryFn: async () => {
+      return await request(route, options);
+    },
+    enabled: options?.prefetch ?? true,
+    refetchOnWindowFocus: options?.refetchOnWindowFocus,
+  });
```

Comment:
In this new approach, we are no longer having the ability to abort requests right? We could modify the `request` to take in a `signal` and use the signal provided by tanstack's useQuery's `queryFn`.

```suggestion
export default function useTanStackQueryInstead<TData>(
  route: QueryRoute<TData>,
  options?: QueryOptions<TData>,
) {
  const { data: response, isLoading } = useQuery({
    queryKey: [route.path, options?.pathParams, options?.query],
    queryFn: async ({ signal }) => {
      return await request(route, options, signal);
    },
    enabled: options?.prefetch ?? true,
    refetchOnWindowFocus: options?.refetchOnWindowFocus,
  });
```